server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -354,20 +353,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             ObjectParser.ValueType.OBJECT
         );
 
-        PARSER.declareInt(
-            ReindexRequest::setMaxDocsValidateIdentical,
-            new ParseField("max_docs", "size").forRestApiVersion(RestApiVersion.equalTo(RestApiVersion.V_7))
-        );
+        PARSER.declareInt(ReindexRequest::setMaxDocsValidateIdentical, new ParseField("max_docs"));
 
-        PARSER.declareInt(
-            ReindexRequest::setMaxDocsValidateIdentical,
-            new ParseField("max_docs").forRestApiVersion(RestApiVersion.onOrAfter(RestApiVersion.V_8))
-        );
         // avoid silently accepting an ignored size.
-        PARSER.declareInt(
-            (r, s) -> failOnSizeSpecified(),
-            new ParseField("size").forRestApiVersion(RestApiVersion.onOrAfter(RestApiVersion.V_8))
-        );
+        PARSER.declareInt((r, s) -> failOnSizeSpecified(), new ParseField("size"));
```

Comment:
The parser should error on unknown from what I can tell. Maybe we should remove the `size` parse field altogether?

Reply:
Yeah not sure if we care about the specific message here or not - I'll let the distrib/indexing team decide that (see 458b0fb71153).